pm/sendermonitor.go

```diff
@@ -313,26 +349,24 @@ func (sm *LocalSenderMonitor) cleanup() {
 }
 
 func (sm *LocalSenderMonitor) redeemWinningTicket(ticket *SignedTicket) (*types.Transaction, error) {
-	maxFloat, err := sm.MaxFloat(ticket.Ticket.Sender)
+	availableFunds, err := sm.availableFunds(ticket.Sender)
 	if err != nil {
 		return nil, err
 	}
 
-	// if max float is zero, there is no claimable reserve left or reserve is 0
-	if maxFloat.Cmp(big.NewInt(0)) <= 0 {
-		if err := sm.QueueTicket(ticket); err != nil {
-			return nil, err
-		}
-		return nil, errors.New("max float is 0")
+	ctx, cancel := context.WithTimeout(context.Background(), sm.cfg.RPCTimeout)
+	gasPrice, err := sm.cfg.SuggestGasPrice(ctx)
+	if err != nil {
+		cancel()
+		return nil, err
 	}
+	cancel()
 
-	// If max float is insufficient to cover the ticket face value, queue
-	// the ticket to be retried later
-	if maxFloat.Cmp(ticket.Ticket.FaceValue) < 0 {
-		if err := sm.QueueTicket(ticket); err != nil {
-			return nil, err
-		}
-		return nil, fmt.Errorf("insufficient max float sender=%v faceValue=%v maxFloat=%v", ticket.Ticket.Sender.Hex(), ticket.Ticket.FaceValue, maxFloat)
+	// We only submit a redemption if availableFunds covers the redemption tx cost
+	// Otherwise, we return an error so we can try the redemption later
+	txCost := new(big.Int).Mul(big.NewInt(int64(sm.cfg.RedeemGas)), gasPrice)
+	if availableFunds.Cmp(txCost) <= 0 {
```

Comment:
This will leave the ticket as the earliest winning ticket for the orchestrator and not mark it as redeemed.

This ticket will then be tried to be redeemed forever, so it seems like we'd need https://github.com/livepeer/go-livepeer/pull/1598 as well for a full robust solution.

Reply:
> This will leave the ticket as the earliest winning ticket for the orchestrator and not mark it as redeemed.

Yep thanks for pointing that out! Worth noting that this is behavior is not introduced by this PR and this PR just changes the condition under which the earliest winning ticket would be left at the head of the queue. That being said, it is worthwhile acknowledging this so we can address this in other PRs such as #1598.